packages/opencensus-instrumentation-http2/test/test-http2.ts

```diff
@@ -0,0 +1,308 @@
+          assert.strictEqual(root.spans.length, 1);
+          assert.ok(root.spans[0].name.indexOf(testPath) >= 0);
+          assert.strictEqual(root.traceId, root.spans[0].traceId);
+          const span = root.spans[0];
+          assertSpanAttributes(span, statusCode, 'GET', host, testPath, null);
+        });
+      });
+    });
+
+    httpErrorCodes.map(errorCode => {
+      it(`should test a child spans for GET requests with http error ${
+             errorCode}`,
+         () => {
+           const testPath = `/${errorCode}`;
+           const requestOptions = {':method': 'GET', ':path': testPath};
+           const options = {name: 'TestRootSpan'};
+
+           return tracer.startRootSpan(options, async root => {
+             await http2Request.get(client, requestOptions).then((result) => {
+               assert.ok(root.name.indexOf('TestRootSpan') >= 0);
+               assert.strictEqual(root.spans.length, 1);
+               assert.ok(root.spans[0].name.indexOf(testPath) >= 0);
+               assert.strictEqual(root.traceId, root.spans[0].traceId);
+
+               const span = root.spans[0];
+               assertSpanAttributes(
+                   span, errorCode, 'GET', host, testPath, null);
+             });
+           });
+         });
+    });
+
+    it('should create multiple child spans for GET requests', () => {
+      const statusCode = 200;
+      const testPath = `/${statusCode}`;
+      const requestOptions = {':method': 'GET', ':path': testPath};
+      const num = 5;
+      const options = {name: 'TestRootSpan'};
+
+      return tracer.startRootSpan(options, async root => {
+        assert.ok(root.name.indexOf('TestRootSpan') >= 0);
+        for (let i = 0; i < num; i++) {
+          await http2Request.get(client, requestOptions).then((result) => {
+            assert.strictEqual(root.spans.length, i + 1);
+            assert.ok(root.spans[i].name.indexOf(testPath) >= 0);
+            assert.strictEqual(root.traceId, root.spans[i].traceId);
+          });
+        }
+        assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+        root.end();
+        assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 1);
+      });
+    });
+
+    it('should not trace requests with \'x-opencensus-outgoing-request\' header',
+       async () => {
+         const statusCode = 200;
+         const testPath = `/${statusCode}`;
+         const requestOptions = {
+           ':method': 'GET',
+           ':path': testPath,
+           'x-opencensus-outgoing-request': 1
+         };
+
+         assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+         await http2Request.get(client, requestOptions).then((result) => {
+           assert.strictEqual(result, `${statusCode}`);
+           assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+         });
+       });
+  });
+
+
+  /** Should intercept incoming requests */
+  describe('patchEmit()', () => {
```

Comment:
We should be testing user-visible behavior here. `patchEmit` is not meant to be user-visible. Can you make this (and other `describe` clauses) a more public-friendly description of what is being tested here?

Reply:
What do you think of changing `'patchEmit()'` to `'Patch incoming requests'` and `'patchRequest()'` to `'Patch outgoing requests'`?

Reply:
Yep, that's an improvement. `Instrumenting incoming requests` would be more accurate, because even patching is not user-visible -- however, instrumenting is (because users will observe traces being reported)

Reply:
Great idea! I'll update that.